src/SignalR/server/SignalR/test/HubConnectionHandlerTests.cs

```diff
@@ -1327,6 +1327,19 @@ public async Task CannotCallDisposeMethodOnHub()
             }
         }
 
+        [Fact]
+        public void CannotHaveGenericMethodOnHub()
+        {
+            using (StartVerifiableLog())
+            {
+                var serviceProvider = HubConnectionHandlerTestUtils.CreateServiceProvider(null, LoggerFactory);
+
+                var exception = Assert.Throws<NotSupportedException>(() => serviceProvider.GetService<HubConnectionHandler<GenericMethodHub>>());
+
+                Assert.Equal("Method 'GenericMethod' is a generic method which is not supported on a Hub.", exception.Message);
```

Comment:
How does this exception reported in a normal app? An error log?

Reply:
```
Application startup exception: System.NotSupportedException: Method 'M' is a generic method which is not supported on a Hub.
   at Microsoft.AspNetCore.SignalR.Internal.DefaultHubDispatcher`1.DiscoverHubMethods() in C:\GitHub\aspnetcore\src\SignalR\server\Core\src\Internal\DefaultHubDispatcher.cs:line 555
   at Microsoft.AspNetCore.SignalR.Internal.DefaultHubDispatcher`1..ctor(IServiceScopeFactory serviceScopeFactory, IHubContext`1 hubContext, Boolean enableDetailedErrors, ILogger`1 logger) in C:\GitHub\aspnetcore\src\SignalR\server\Core\src\Internal\DefaultHubDispatcher.cs:line 37
   at Microsoft.AspNetCore.SignalR.HubConnectionHandler`1..ctor(HubLifetimeManager`1 lifetimeManager, IHubProtocolResolver protocolResolver, IOptions`1 globalHubOptions, IOptions`1 hubOptions, ILoggerFactory loggerFactory, IUserIdProvider userIdProvider, IServiceScopeFactory serviceScopeFactory) in C:\GitHub\aspnetcore\src\SignalR\server\Core\src\HubConnectionHandler.cs:line 74
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   etc...
```

Reply:
This won't break hubs with private or internal generic methods, right? Even if that's the case, technically this is a breaking change since it's possible that there are public generic Hub methods that were only called indirectly. Still think it's a worthwhile change. Not sure whether it warrants a breaking change announcement.

Reply:
Nvm. I just looked at the issue again, and saw it was crashing at startup even previously. Carry on 😄